aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java

```diff
@@ -243,15 +234,17 @@ public void testDropTable() {
                     .key(DynamoDbCatalog.tablePrimaryKey(tableIdentifier))
                     .build())
             .hasItem());
-    AssertHelpers.assertThrows(
-        "metadata location should be deleted",
-        NoSuchKeyException.class,
-        () ->
-            s3.headObject(
-                HeadObjectRequest.builder()
-                    .bucket(testBucket)
-                    .key(metadataLocation.substring(testBucket.length() + 6)) // s3:// + end slash
-                    .build()));
+    Assertions.assertThatThrownBy(
+            () ->
+                s3.headObject(
+                    HeadObjectRequest.builder()
+                        .bucket(testBucket)
+                        .key(
+                            metadataLocation.substring(
+                                testBucket.length() + 6)) // s3:// + end slash
+                        .build()))
+        .as("metadata location should be deleted")
+        .isInstanceOf(NoSuchKeyException.class);
```

Comment:
we need a `hasMessageXyZ(...)` check here.

Reply:
added